snap/squashfs/squashfs.go

```diff
@@ -273,6 +275,131 @@ func (s *Snap) ReadFile(filePath string) (content []byte, err error) {
 	return content, nil
 }
 
+var sdVer int
+
+func init() {
+	// this also avoids calling systemctl all the time which perturbs
+	// some unrelated tests
+	v, err := systemd.Version()
+	if err != nil {
+		v = 0
+	}
+	sdVer = v
+}
+
+var (
+	sysGeteuid = sys.Geteuid
+)
+
+func sandboxParams(sdVer int) (params []string) {
+	if sysGeteuid() != 0 {
+		return nil
+	}
+	// effetive user is 0, use systemd-run to sandbox unsquashfs
+	params = []string{
+		"--property=KeyringMode=private",
+		"--property=PrivateTmp=true",
+		// xxx CAP_DAC_OVERRIDE is mainly possibly needed for
+		// prepare-image run under sudo
+		"--property=CapabilityBoundingSet=CAP_DAC_OVERRIDE",
+		"--property=AmbientCapabilities=CAP_DAC_OVERRIDE",
+		"--property=NoNewPrivileges=true",
+		"--property=SecureBits=",
+		"--property=ProtectSystem=strict",
+		"--property=ProtectHome=true",
+		"--property=ProtectKernelTunables=true",
+		"--property=ProtectControlGroups=true",
+		"--property=ProtectKernelModules=true",
+		"--property=PrivateDevices=true",
+		"--property=MemoryDenyWriteExecute=true",
+		"--property=RestrictSUIDSGID=true",
+		"--property=SystemCallFilter=@default @basic-io @signal @file-system @chown @process mprotect",
+		"--property=SystemCallErrorNumber=EPERM",
+		"--property=MemoryMax=16M",
+	}
+
+	// see https://github.com/systemd/systemd/blob/main/NEWS
+
+	if sdVer >= 240 {
+		params = append(params, "--service-type=exec")
+	}
+	if sdVer >= 242 {
+		params = append(params, "--property=ProtectHostname=true")
+	}
+	if sdVer >= 244 {
+		params = append(params, "--property=ProtectKernelLogs=true")
+	}
+	if sdVer >= 245 {
+		params = append(params, "--property=ProtectClock=true")
+	}
+
+	return params
+}
+
+// SaferReadFile returns the content of a single file inside a squashfs snap, but it does invoking unsquashfs under a sandbox to handle not yet verified snap files, there is a limit below 16M on the content size.
```

Comment:
can this comment please be filled to say 80 columns or similar to make it easier to read?